src/internal/header.rs

```diff
@@ -1,25 +1,30 @@
 //! RPM package headers
 
 use super::{tag::Tag, td::TagData};
-use crate::{License, Package, Version};
 use std::mem;
 
 /// RPM package header
-pub(crate) struct Header(*mut librpm_sys::headerToken_s);
+pub(crate) struct Header(*mut librpm_sys::headerToken_s); // librpm_sys::Header == *mut headerToken_s
 
-impl Header {
-    /// Create a new header from an `librpm_sys::Header`.
-    pub(crate) fn new(ffi_header: librpm_sys::Header) -> Self {
+/// Create a new header from an `librpm_sys::Header`.
+impl From<librpm_sys::Header> for Header {
+    fn from(ffi_header: librpm_sys::Header) -> Self {
         unsafe {
```

Comment:
This is very much unsafe code as it's dealing with raw pointers and probably shouldn't be part of the public API, unless there are very good reasons why an end-user of the crate would be mucking about with `librpm_sys::Header` pointers.

 ```suggestion
impl Header {
    pub(crate) unsafe fn from_ptr(ptr: librpm_sys::Header) -> Self {
        unsafe {
```

Reply:
The `Header` type is currently restricted to `pub(crate)` so the trait impl wouldn't be part of the API, at least not as it currently stands.  

But if `Header` needs or might-need to be part of the public API eventually then I completely agree.

Reply:
Even if it's internal to the crate only, functions which can lead to unsoundness need to be annotated as `unsafe`, and ideally also have a `/// SAFETY: ...` comment explaining how to use them soundly.

Some examples of unsoundness here are potential UAFs, and a SAFETY comment would briefly describe librpm's reference counting model and how the reference count must be at least 1 at the time this is called.